blogposts/announcing-sourcegraph-3.1.md

```diff
@@ -0,0 +1,100 @@
+
+Improving code intelligence is a priority for us in  2019 so expect regular improvement updates in future releases. Want to improve support for your favorite language(s)? [Create an issue](https://github.com/sourcegraph/sourcegraph/issues/new?assignees=&labels=&template=feature_request.md&title=Language%20extension%20improvement%20for).
+
+## Site admin improvements
+
+Administering Sourcegraph in an enterprise SSO environment is now easier thanks to a couple of small, but important features:
+
+- [Querying users by email address](https://sourcegraph.com/api/console#%7B%22query%22%3A%22%7B%5Cn%20%20user(email%3A%20%5C%22ryan%40sourcegraph.com%5C%22)%20%7B%5Cn%20%20%20%20id%5Cn%20%20%7D%5Cn%7D%5Cn%22%7D).
+- [Disabling the changing of usernames](https://docs.sourcegraph.com/admin/config/critical_config#reference).
+
+Making configuration changes through the JSON editor is now easier too with easy to access documentation in the editor itself.
+
+<!-- TODO (ryan): GIF showing editor using the help tool tip with an item that requires a choice of values -->
+
+## Improved configuration management and documentation
+
+Sourcegraph 3.0 introduced the [management console](https://about.sourcegraph.com/blog/sourcegraph-3.0#management-console-and-site-config-improvements) for editing critical configuration to allow recovery from misconfiguration in the site admin.
+
+We've [updated our documentation](https://docs.sourcegraph.com/admin/config) to reflect this change, separating configuration into two sections:
+
+- [Critical configuration](https://docs.sourcegraph.com/admin/config/critical_config).
+- [Site configuration](https://docs.sourcegraph.com/admin/config/site_config).
+
+Reference documentation for [site configuration](https://docs.sourcegraph.com/admin/config/site_config#reference) and [critical configuration](https://docs.sourcegraph.com/admin/config/critical_config#reference) values are now rendered using the schema files themselves to ensure they'll never get out of date.
+
+## New Sourcegraph extension APIs
+
+The [Sourcegraph extension API](https://docs.sourcegraph.com/extensions/authoring) continues to evolve with two new APIs:
+
+- [Exposing text selection to extensions](https://github.com/sourcegraph/sourcegraph/issues/702).
+- [APIs for listening to active text editor changes and selections](https://github.com/sourcegraph/sourcegraph/issues/2026).
+
+This means the `[Git extras Sourcegraph extension](https://sourcegraph.com/extensions?query=git-extras) works better than ever. Want this functionality on GitHub too? Just install the [Sourcegraph browser extension](https://docs.sourcegraph.com/integration/browser_extension).
+
+<!-- TODO (ryan): Gif of git-extras extension -->
+
+## Switching to scheduled releases on the 20th of each month
+
+We received feedback from many of you asking for a more regular and predictable release schedule and as a result, Sourcegraph will now be released on the 20th of each month 🎉.
```

Comment:
The first half of this sentence is boring, and the whole sentence is a bit run-on.

Ideas for improvement:

- Describe up front the release schedule and link to the internal release docs  at https://docs.sourcegraph.com/dev/releases
- Remind people that we have released N versions of Sourcegraph in the last M months, about one per month (since Dec 2017!). This is awesome and it shows that we are great at regularly releasing and making continual progress.
- Credit GitLab for inspiration and link to their blog post about how they release on the 22nd of each month
- Encourage people to provide feedback as quickly as possible so we can incorporate it in the next release!

Reply:
@sqs I've [rewritten the fixed release](https://github.com/sourcegraph/about/pull/83/commits/f6ae50b31d8381865c97cb76448a5220a7b572bf) section  but am not sure if it's what you had in mind.

Reply:
Looks great high level. Haven’t thoroughly reviewed it